### DISCUSSION TITLE
Allow CI for private pull requests

### DISCUSSION THREAD

**Discussion by glye** [community/community#39859] [Code Security]
### Select Topic Area

Product Feedback

### Body

I understand why CI / workflows currently do not run for private pull requests in security advisories. Running them would require sending sensitive data to 3rd party organisations, an obvious risk (1).

However, this complete block raises its own risk (2): In a large project with many repos and complex CI, it is difficult to do all the CI tests locally. The result can be that blocking flaw only becomes apparent *after* the private pull request has been merged into a public repo, thus exposing the vulnerability *before* the fix is completely functional. This leaves users vulnerable to attack for however much time of frantic fixing is required to unblock it, and can result in further flaws because of the desperate urgency involved.

It also lessens the value of CI on GitHub in general. If we have to make the effort of maintaining our own local CI for security fixes, we could use the same for regular commits also, and save the money we currently spend on external CI tests.

I believe risk 2 is worse than risk 1. GitHub should allow organisations to make this choice themselves. Please make it possible to unblock CI on private pull requests, per organisation or repository, with the adequate warnings and disclaimers of course. It's the least dangerous solution!

If you do this and a 3rd party CI provider betrays our trust, they will likely be found out and lose their customers. It is very much in their own interest to maintain strict discipline on use of their clients' data. And everyone understands GitHub would not be to blame in such a breach.

**Comment by KateCatlin** [comment]
Hi @glye, thank you so much for taking the time to post about this. It's something that's been on our minds and is on our roadmap. 

I'm a PM here at GitHub and I'd love to chat with you about the specifics of your user requirements. Would you be open to a 30-minute call to discuss this? Happy to send you a gift card as gratitude for your time: https://calendly.com/security-advisories-ux-calls/30min?month=2022-12

**Reply by glye** [reply]
@KateCatlin Great, I'd be happy to. Maybe we can fit it into the call already planned for tomorrow? Ref: https://github.com/orgs/community/discussions/14021#discussioncomment-4218666

**Reply by KateCatlin** [reply]
@glye it seems you're giving us all sorts of good feedback lately :) 
Let's try to discuss both!
